examples/transactions.c

```diff
@@ -184,135 +154,137 @@ create_transactional_producer (const rd_kafka_topic_partition_t *rktpar) {
         if (error)
                 fatal_error("init_transactions()", error);
 
-
-        /* Begin a new transaction */
-        error = rd_kafka_begin_transaction(rk);
-        if (error)
-                fatal_error("begin_transaction()", error);
-
         return rk;
 }
 
 
 /**
- * @brief Abort the current transaction and destroy the producer.
+ * @brief Rewind consumer's consume position to the last committed offsets
+ *        for the current assignment.
  */
-static void destroy_transactional_producer (rd_kafka_t *rk) {
+static void rewind_consumer (rd_kafka_t *consumer) {
+        rd_kafka_topic_partition_list_t *offsets;
+        rd_kafka_resp_err_t err;
         rd_kafka_error_t *error;
+        int i;
 
-        fprintf(stdout, "%s: aborting transaction and terminating producer\n",
-                rd_kafka_name(rk));
+        /* Get committed offsets for the current assignment, if there
+         * is a current assignment. */
+        err = rd_kafka_assignment(consumer, &offsets);
+        if (err) {
+                fprintf(stderr, "No current assignment to rewind: %s\n",
+                        rd_kafka_err2str(err));
+                return;
+        }
 
-        /* Abort the current transaction, ignore any errors
-         * since we're terminating the producer anyway. */
-        error = rd_kafka_abort_transaction(rk, -1);
-        if (error) {
-                fprintf(stderr,
-                        "WARNING: Ignoring abort_transaction() error since "
-                        "producer is being destroyed: %s\n",
-                        rd_kafka_error_string(error));
-                rd_kafka_error_destroy(error);
+        if (offsets->cnt == 0) {
+                fprintf(stderr, "No current assignment to rewind\n");
+                rd_kafka_topic_partition_list_destroy(offsets);
+                return;
         }
 
-        rd_kafka_destroy(rk);
-}
+        /* Note: Timeout must be lower than max.poll.interval.ms */
```

Comment:
is that enforced? (it doesn't seem to be directly in `rd_kafka_committed` but may be further in, didn't look).

Reply:
Nope, see next comment.